vtm.js

```diff
@@ -326,4 +476,122 @@ app.component('stat', {
     </div>`
 });
 
+/**
+  * displays inputs for name and age
+  * displays selects for clan, generation, faction
+  */
+app.component('character-info', {
+  props: ['bio', 'clans'],
+  
+  template: `
+    <div class="char-info">
+      <input 
+        class="heading1"
+        type="text"
+        v-model="bio.name" 
+        :placeholder="bio.defaultName"
+        />
+      <b>Age:</b>
+      <input
+        :style= "{width: 7 + 'ch'}"
+        v-model="bio.age"
+        type="number"
+        :placeholder = '123'/>
+      <b>Generation:</b>
+      <select 
+        v-model="bio.generation">
+        <option disabled value=null>-none-</option>
+        <option>15th</option>
+        <option>14th</option>
+        <option>13th</option>
+      </select>
+      <b>Clan:</b>
+      <select v-model="bio.clan">
+        <option disabled value="">Pick a Clan</option>
+        <option 
+          v-for="clan in clans"
+          :key = "clan.id"
+          :value="clan">
+          {{clan.id}}
+        </option>
+      </select>
+      <b>Faction:</b>
+      <select v-model="bio.faction">
+        <option disabled value="">Choose Your Coterie's Faction</option>
+        <option>Cammarilla</option>
+        <option>etc</option>
+        <option>C</option>
+      </select>
+      </div>`
+});
+
+app.component('discipline-section', {
+  mixins:[statSectionMixin],
+  props:['clan'],
+  computed: {
+    /**
+     * @returns {Array} of numbers describing how many points are currently assigned
+     * for example [0,9,0,0,0,0] means there are 9 attributes with value 1
+     * while [0,1,4,3,1,0] means 1 attribute with value 1, 4 attributes with value 2,
+     * 3 attributes with value 3, 1 attribute with value 4 and 0 attributes with value 5
+     */
+    allocatedResources() {
```

Comment:
toto je stejne vo viacero componentach, nedalo by sa to tiez vzat s Mixin?

Reply:
discipliny su napisane inac